cli/cmd/inject.go

```diff
@@ -278,71 +287,109 @@ func (options *injectOptions) fetchConfigsOrDefault() (*config.All, error) {
 }
 
 // overrideConfigs uses command-line overrides to update the provided configs
-func (options *injectOptions) overrideConfigs(configs *config.All) {
+func (options *injectOptions) overrideConfigs(configs *config.All, overrideAnnotations map[string]string) {
 	if len(options.ignoreInboundPorts) > 0 {
 		configs.Proxy.IgnoreInboundPorts = toPorts(options.ignoreInboundPorts)
+		overrideAnnotations[k8s.ProxyIgnoreInboundPortsAnnotation] = parsePorts(configs.Proxy.IgnoreInboundPorts)
 	}
 	if len(options.ignoreOutboundPorts) > 0 {
 		configs.Proxy.IgnoreOutboundPorts = toPorts(options.ignoreOutboundPorts)
+		overrideAnnotations[k8s.ProxyIgnoreOutboundPortsAnnotation] = parsePorts(configs.Proxy.IgnoreOutboundPorts)
 	}
 
 	if options.proxyAdminPort != 0 {
 		configs.Proxy.AdminPort = toPort(options.proxyAdminPort)
+		overrideAnnotations[k8s.ProxyAdminPortAnnotation] = parsePort(configs.Proxy.AdminPort)
 	}
 	if options.proxyControlPort != 0 {
 		configs.Proxy.ControlPort = toPort(options.proxyControlPort)
+		overrideAnnotations[k8s.ProxyControlPortAnnotation] = parsePort(configs.Proxy.ControlPort)
 	}
 	if options.proxyInboundPort != 0 {
 		configs.Proxy.InboundPort = toPort(options.proxyInboundPort)
+		overrideAnnotations[k8s.ProxyInboundPortAnnotation] = parsePort(configs.Proxy.InboundPort)
 	}
 	if options.proxyOutboundPort != 0 {
 		configs.Proxy.OutboundPort = toPort(options.proxyOutboundPort)
+		overrideAnnotations[k8s.ProxyOutboundPortAnnotation] = parsePort(configs.Proxy.OutboundPort)
 	}
 
-	if options.dockerRegistry != "" {
```

Comment:
We still need to check for image name even if the registry option isn't provided.